acceptance/tests/ticket_3961_puppet_ca_should_produce_certs.rb

```diff
@@ -32,5 +32,9 @@
   on agent, "test -f #{scratch}/ssl/certs/#{target}.pem"
 
   step "verify the private key for #{target} exists"
-  on agent, "grep 'BEGIN RSA PRIVATE KEY' #{scratch}/ssl/private_keys/#{target}.pem > /dev/null 2>&1"
+  if (on(agent, facter("find fips_enabled")).stdout =~ /true/)
+    on agent, "grep 'BEGIN PRIVATE KEY' #{scratch}/ssl/private_keys/#{target}.pem > /dev/null 2>&1"
+  else
+    on agent, "grep 'BEGIN RSA PRIVATE KEY' #{scratch}/ssl/private_keys/#{target}.pem > /dev/null 2>&1"
+  end
```

Comment:
This is surprising. Based on https://stackoverflow.com/questions/20065304/what-is-the-differences-between-begin-rsa-private-key-and-begin-private-key it sounds like the FIPS enabled version is PKCS#8 and the other is PKCS#1. Why does FIPS mode change that?

Reply:
Yeah. I too could not find anything indicating it is supposed to. Attempting to generate key pairs using openssl cmd, irrespective of FIPS mode, always seem to generate PKCS#1 format key. And neither does it seem related to the version of openssl.  Strangely there are comments suggesting openssl switched to using pkcs8 format after version 0.9.8 but that does not to be the case.  
It does change behavior however based on fips mode when attempting to create it programmatically using "OpenSSL::PKey::RSA.new(keysize)". I will investigate it later but the PKCS#1 form does not seem to have any ill-effects.